src/Connect.js

```diff
@@ -1,68 +1,441 @@
+ *    claim: {
+ *      "Citizen of city X": {
+ *        "Allowed to vote": true,
+ *        "Document": "QmZZBBKPS2NWc6PMZbUk9zUHCo1SHKzQPPX4ndfwaYzmPW"
+ *      }
+ *    },
+ *    sub: "2oTvBxSGseWFqhstsEHgmCBi762FbcigK5u"
+ *  }
+ *  credentials.createVerificationRequest(unsignedClaim).then(jwt => {
+ *    ...
+ *  })
+ *
+ *  @param    {Object}      reqObj                 object with request params
+ *  @param    {Object}      reqObj.unsignedClaim   an object that is an unsigned claim which you want the user to attest
+ *  @param    {String}      reqObj.sub             the DID which the unsigned claim is about
+ *  @param    {String}      [id='signClaimReq']    string to identify request, later used to get response
+ */
+  createVerificationRequest(reqObj, id='signClaimReq') {
+    this.credentials.createVerificationRequest(reqObj.unsignedClaim, reqObj.sub, callbackUrl: this.genCallback(), this.did)
+                    .then(jwt => his.request(jwt, id))
+  }
+
+  /**
+ *  Creates a [Selective Disclosure Request JWT](https://github.com/uport-project/specs/blob/develop/messages/sharereq.md)
+ *
+ *  @example
+ *  const req = { requested: ['name', 'country'],
+ *                callbackUrl: 'https://myserver.com',
+ *                notifications: true }
+ *  credentials.requestDisclosure(req).then(jwt => {
+ *      ...
+ *  })
+ *
+ *  @param    {Object}             [params={}]           request params object
+ *  @param    {Array}              params.requested      an array of attributes for which you are requesting credentials to be shared for
+ *  @param    {Array}              params.verified       an array of attributes for which you are requesting verified credentials to be shared for
+ *  @param    {Boolean}            params.notifications  boolean if you want to request the ability to send push notifications
+ *  @param    {String}             params.callbackUrl    the url which you want to receive the response of this request
+ *  @param    {String}             params.network_id     network id of Ethereum chain of identity eg. 0x4 for rinkeby
+ *  @param    {String}             params.accountType    Ethereum account type: "general", "segregated", "keypair", "devicekey" or "none"
+ *  @param    {Number}             params.expiresIn      Seconds until expiry
+ *  @param    {String}            [id='disclosureReq']    string to identify request, later used to get response
+ *  @return   {Promise<Object, Error>}                   a promise which resolves with a signed JSON Web Token or rejects with an error
+ */
+  requestDisclosure (reqObj, id='disclosureReq') {
+    this.credentials.requestDisclosure(Object.assign(params, {callbackUrl: this.genCallback()}), reqObj.expiresIn)
+                    .then(jwt => this.request(jwt, id))
+  }
+
+  /**
+  *  Create a credential about connnected user
+  *
+  *  @example
+  *  credentials.attest({
+  *   sub: '5A8bRWU3F7j3REx3vkJ...', // uPort address of user, likely a MNID
```

Comment:
We should default this to the did of the user if present, but allow it to be overwritten.